### PR TITLE
Several improvements to the web UI

### LINKED ISSUE #5599: Use contains instead of regex for UI search

### LINKED ISSUE #5675: Add copy query button to the Query Details page
It would be helpful to be able to quickly copy the query text to the clipboard.

### LINKED ISSUE #5971: UI active worker count is incorrect
The coordinator should not be included as a "worker" unless `node-scheduler.include-coordinator` is true.

### LINKED ISSUE #6226: Add stage state to stage summary in query info page
Also if possible add sum of output buffer size since that shows which stages are blocked on output.

### PR DESCRIPTION
**Pull request by raghavsethi**
Fixes #6226 
Fixes #5971 
Fixes #5599 
Fixes #5675

### CODE REVIEW

**Comment by martint** [comment]
Does this include the fix to only show % progress when all splits have finished scheduling?

**Comment by raghavsethi** [comment]
@martint yes, it does - it's in the commit `Create field for human readable query status in QueryInfo`

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 1]
```diff
@@ -728,9 +731,27 @@ var QueryDetail = React.createClass({
             return <button className="btn btn-info live-button" onClick={ this.handleStageRefreshClick }>Auto-Refresh: Off</button>
         }
     },
+    renderTaskFilterListItem: function(taskFilter, taskFilterText) {
+        return (
+            <li><a href="#" className={ this.state.taskFilter == taskFilter ? "selected" : ""} onClick={this.handleTaskFilterClick.bind(this, taskFilter) }>{ taskFilterText }</a></li>
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query.js]: Should there be a space after `onClick={`?

File: presto-main/src/main/resources/webapp/assets/query-list.js [hunk 2]
```diff
@@ -257,11 +257,12 @@ var QueryList = React.createClass({
         }
         else {
             return stateFilteredQueries.filter(function(query) {
-                var re = (".*" + searchString + ".*").toLowerCase();
-                return (query.queryId.toLowerCase().match(re) ||
-                    query.session.user.toLowerCase().match(re) ||
-                    query.session.source.toLowerCase().match(re) ||
-                    query.query.toLowerCase().match(re));
+                var term = searchString.toLowerCase();
+                return (query.queryId.toLowerCase().indexOf(term) != -1 ||
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query-list.js]: Maybe add a helper function `containsLowercasedString`?

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 3]
```diff
@@ -648,7 +648,7 @@ var QueryDetail = React.createClass({
                 lastByteInput: parseDataSize(query.queryStats.processedInputDataSize),
 
                 initialized: true,
-                ended: query.state != "RUNNING",
+                ended: query.finalQueryInfo,
```
Comment (raghavsethi) [inline presto-main/src/main/resources/webapp/assets/query.js]: This is a boolean.

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 4]
```diff
@@ -802,33 +802,21 @@ var QueryDetail = React.createClass({
...
-                        summary += " (" + query.queryStats.blockedReasons.join() + ")";
-                    }
-                }
-                else {
-                    summary = (progress == 0 ? summary : summary + " (" + progress + "%" + ")");
-                }
-                break;
+
+        var progress = 0;
+        if (query.scheduled && query.state == "RUNNING" && query.queryStats.totalDrivers > 0 && query.queryStats.completedDrivers > 0) {
```
Comment (raghavsethi) [inline presto-main/src/main/resources/webapp/assets/query.js]: Almost - it's one has query.queryStats.x and the other has query.x

File: presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java [hunk 5]
```diff
@@ -263,6 +268,42 @@ public boolean isFinalQueryInfo()
     }
 
     @JsonProperty
+    public String getHumanReadableState()
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java]: Why is this in QueryInfo? It presentation-specific logic, so it probably belongs in the UI or, worst case, in the jaxrs resource.

File: presto-main/src/main/java/com/facebook/presto/server/BasicQueryStats.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+        this.elapsedTime = elapsedTime;
+        this.executionTime = executionTime;
+        this.cpuTime = cpuTime;
+        this.fullyBlocked = fullyBlocked;
+        this.blockedReasons = ImmutableSet.copyOf(requireNonNull(blockedReasons, "blockedReasons is null"));
+        this.cumulativeMemory = cumulativeMemory;
+        this.totalMemoryReservation = totalMemoryReservation;
+        this.peakMemoryReservation = peakMemoryReservation;
+
+        checkArgument(totalDrivers >= 0, "totalDrivers is less than zero");
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/BasicQueryStats.java]: Make all these "is negative" to match `QueryStats`

File: presto-main/src/main/resources/webapp/assets/query-list.js [hunk 7]
```diff
@@ -52,89 +52,53 @@
...
             <div className="col-xs-12 tinystat-row">
                 <span className="tinystat" data-toggle="tooltip" data-placement="top" title="Completed splits">
                     <span className="glyphicon glyphicon-ok" style={ GLYPHICON_HIGHLIGHT }></span>&nbsp;&nbsp;
-                    { completedDrivers }
+                    { query.queryStats.completedDrivers }
                 </span>
                 <span className="tinystat" data-toggle="tooltip" data-placement="top" title="Running splits">
                     <span className="glyphicon glyphicon-play" style={ GLYPHICON_HIGHLIGHT }></span>&nbsp;&nbsp;
-                    { runningDrivers }
+                    { query.state == "FINISHED" || "FAILED" ? 0 : query.queryStats.runningDrivers }
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query-list.js]: Wrong syntax. Also add parenthesis
```js
(query.state == "FINISHED" || query.state == "FAILED") ? 0 : query.queryStats.runningDrivers

File: presto-main/src/main/resources/webapp/assets/query-list.js [hunk 8]
```diff
@@ -52,89 +52,53 @@
...
                 </span>
                 <span className="tinystat" data-toggle="tooltip" data-placement="top" title="Queued splits">
                     <span className="glyphicon glyphicon-pause" style={ GLYPHICON_HIGHLIGHT }></span>&nbsp;&nbsp;
-                    { queuedDrivers }
+                    { query.state == "FINISHED" || "FAILED" ? 0 : query.queryStats.queuedDrivers }
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query-list.js]: Same

File: presto-main/src/main/resources/webapp/assets/query-list.js (hunk 7 above)
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query-list.js]: Maybe add a helper function `isDoneState(query.state)`

File: presto-main/src/main/resources/webapp/assets/utils.js [hunk 9]
```diff
@@ -85,6 +63,61 @@ function getQueryStateColor(state, errorType, errorCode)
...
+                return "EXTERNAL ERROR";
+        }
+    }
+
+    return query.state;
+}
+
+function getProgressBarPercentage(query)
+{
+    if (query.scheduled && query.state == "RUNNING" && query.queryStats.totalDrivers > 0 && query.queryStats.completedDrivers > 0) {
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/utils.js]: Add a helper for this expression, it is repeated below

File: presto-main/src/main/resources/webapp/assets/utils.js [hunk 10]
```diff
@@ -85,6 +63,61 @@ function getQueryStateColor(state, errorType, errorCode)
...
+        return Math.round((query.queryStats.completedDrivers * 100.0) / query.queryStats.totalDrivers);
+    }
+
+    return 100;
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/utils.js]: Add a comment explaining why it defaults to 100

**Comment by electrum** [comment]
Add tests that construct `BasicQueryInfo` and `BasicQueryStats` then call the getters, to make sure none of the fields are swapped
